scalalib/src/mill/scalalib/cosmopolitan/CosmopolitanModule.scala

```diff
@@ -0,0 +1,131 @@
+package mill.scalalib.cosmopolitan
+
+import mill.api.{PathRef, Result}
+import mill.define.{Target => T, _}
+import mill.scalalib._
+
+trait CosmopolitanModule extends mill.Module with AssemblyModule {
+  def forkArgs(argv0: String): Task[Seq[String]] = Task.Anon { Seq[String]() }
+
+  def cosmoccVersion: T[String] = Task { "" }
+
+  def cosmocc: T[PathRef] = Task(persistent = true) {
+    val version = if (cosmoccVersion().isEmpty) "" else s"-${cosmoccVersion()}"
+    os.write(
+      Task.dest / "cosmocc.zip",
+      requests.get.stream(s"https://cosmo.zip/pub/cosmocc/cosmocc${version}.zip")
+    )
+
+    os.call(("unzip", Task.dest / "cosmocc.zip", "-d", Task.dest / "cosmocc"))
+    PathRef(Task.dest / "cosmocc" / "bin" / "cosmocc")
+  }
```

Comment:
The idea behind the `persistent = true` is, that `T.dest` doesn't get automatically removed. We still need to check, if the wanted zip-file is present and only download it if not.